pymc/testing.py

```diff
@@ -574,7 +574,7 @@ def check_icdf(
     # Test pymc and scipy distributions match for values and parameters
     # within the supported domain edges (excluding edges)
     domains = paramdomains.copy()
-    domain = Domain([0, 0.1, 0.5, 0.75, 0.95, 0.99, 1])  # Values we test the icdf at
+    domain = Domain([0, 0.1, 0.5, 0.75, 0.95, 0.99, 0.9999, 1])  # Values we test the icdf at
```

Comment:
Do the tests fail with just 3 9's? I am wary of testing such an extreme value by default because it puts a high burden on future icdfs on being very accurate for extreme values. We know these things must always break at some point, specially for unbounded distributions.

```suggestion
    domain = Domain([0, 0.1, 0.5, 0.75, 0.95, 0.99, 0.999, 1])  # Values we test the icdf at
```

Reply:
If you want a test that covers the changes in this PR, you can add something specific where you just evaluate the icdf at the values that used to fail.

Reply:
> If you want to cover the changes in this PR, you can add a small test where you just evaluate the icdf at the values that used to fail for the two distributions
To clarify my understanding, you mean I add the specific test for those distributions that fail and not something general that applies to all future distributions right?

Reply:
> Do the old implementations fail with just 3 9's? I am wary of testing such an extreme value by default because it puts a high burden on future icdfs on being very accurate for extreme values. We know these things must always break at some point, specially for unbounded distributions.

I added `0.9999`, since as outlined in https://github.com/pymc-devs/pymc/issues/6670,
for a Geometric distribution with `p=0.99`, and CDF value `0.9999` whose ICDF is expected to be `2` but it gives `3` as the result. It works fine where the ICDF of `0.99` is `1`. To summarize this in a table:

`p=0.99`
| Value | CDF | returned ICDF | expected ICDF |

Reply:
I am guessing it will pass the test for `0.999`, whose ICDF is still `1`. Instead of this I think I can remove this value altogether, and instead add the other type of test you were suggesting, that is `ICDF(CDF(k)) = k` for discrete distributions as I have added for the continuous cases.

Reply:
> To clarify my understanding, you mean I add the specific test for those distributions that fail and not something general that applies to all future distributions right?

Yes that's what I suggest. Also note that instead of tweaking the icdf value we can tweak the domain of the distribution parameters that is tested. Probably testing more extreme Geometric `p` values would also have shown the error?

Regardless, a separate test is also fine

Reply:
Thanks @ricardoV94 .

I have decided to do this, let me know if it makes sense:

With my latest commit, I have removed the value `0.9999` in the test for icdf to not burden all future ICDF code regarding high accuracy.

`p=0.99` is already used, as defined by `Unit` in `testing.py`.

The ICDF tests will instead be covered by the self-consistency check for ICDF of discrete distributions, which I will add in this PR https://github.com/pymc-devs/pymc/pull/6642 . (Note that the self-consistency check for ICDF of _continuous_ distributions is already added in https://github.com/pymc-devs/pymc/pull/6642, and the _discrete_ self consistency check is blocked on the current PR because of this accuracy issue. Thus I will add it to https://github.com/pymc-devs/pymc/pull/6642 after this current PR is closed.)

Reply:
Sounds good